File: codeforces/1145A-thanos/1145A-thanos-naive.py


def isOrdered(arr):
    for i in range(len(arr) - 1):
        if arr[i] > arr[i+1]:
            return False
    return True


def solve(arr):
    if isOrdered(arr):
        return len(arr)

    a = solve(arr[:len(arr)//2])
    b = solve(arr[len(arr)//2:])

    if a > b:
        return a

    return b


if __name__ == '__main__':

    l = int(input())

    arr = [int(a) for a in input().split()]

    print(solve(arr))

File: codeforces/1145A-thanos/1145A-thanos-naive.go
package main

import (
	"fmt"
)

func IsOrdered(arr []int, arrLen int) bool {
	for i := 0; i < arrLen-1; i++ {
		if arr[i] > arr[i+1] {
			return false
		}
	}
	return true
}

func Solve(arr []int, arrLen int) int {
	if IsOrdered(arr, arrLen) {
		return arrLen
	}

	a := Solve(arr[:arrLen/2], arrLen/2)
	b := Solve(arr[arrLen/2:], arrLen/2)

	if a > b {
		return a
	}
	return b
}

func main() {
	var length int
	fmt.Scanf("%d", &length)
	arr := make([]int, length)
	for i := 0; i < length; i++ {
		fmt.Scanf("%d", &arr[i])
	}

	fmt.Println(Solve(arr, length))
}
